_source/_posts/2022-10-20-spring-vault.md

```diff
@@ -0,0 +1,642 @@
+            "okta.oauth2.issuer":"hmac-sha256:70502e5cd54d12004df6836d0ff8d045515e4207607d984da14c0aaca5400c0a"
+         },
+         "options":{
+
+         }
+      },
+      "remote_address":"127.0.0.1",
+      "remote_port":58914
+   }
+}
+```
+
+Let's assume you don't want to configure the root token in the `vault-demo-app`. You can instead create a policy granting read permissions on the path where the secrets were stored. Go to the Vault Web UI at `http://localhost:8200` and log in with the root token (`00000000-0000-0000-0000-000000000000`).
+
+{% img blog/spring-vault-update/vault-web-ui.png alt:"Vault web UI" width:"650" %}{: .center-image }
+
+Next, go to **Policies** and **Create ACL policy**. Create a `vault-demo-app-policy` with the following capabilities:
+
+```json
+path "secret/data/vault-demo-app" {
+  capabilities = [ "read" ]
+}
+
+path "secret/data/vault-demo-app,dev" {
+  capabilities = [ "read" ]
+}
+
+path "secret/data/application" {
+  capabilities = [ "read" ]
+}
+
+path "secret/data/application,dev" {
+  capabilities = [ "read" ]
+}
+```
+
+All the paths above will be requested by the config server to provide configuration for the `vault-demo-app` when it starts with the `dev` profile active.
+
+{% img blog/spring-vault-update/vault-policy.png alt:"Vault policy section" width:"800" %}{: .center-image }
+
+Now, go back to the container command line, and create a token with the `vault-demo-app-policy`.
+
+```shell
+vault token create -policy=vault-demo-app-policy
+```
+
+```shell
+WARNING! The following warnings were returned from Vault:
+
+  * Endpoint ignored these unrecognized parameters: [display_name entity_alias
+  explicit_max_ttl num_uses period policies renewable ttl type]
+
+Key                  Value
+---                  -----
+token                hvs.CAESIKd9pYyc9xesiqmwvepdGiHlPEB53By...
+token_accessor       2pzRXmGhxChobbMtqeM5zZzM
+token_duration       768h
+token_renewable      true
+token_policies       ["default" "vault-demo-app-policy"]
+identity_policies    []
+policies             ["default" "vault-demo-app-policy"]
+```
+
+**NOTE**: I could not find documentation about the warning _Endpoint ignored these unrecognized parameters_. It seems `vault CLI` is sending default parameters not required by the target API in this case. The command equivalent API call can be displayed using the `-output-curl-string` flag after the subcommand, for example:
+
+```shell
+vault token create -output-curl-string -policy=vault-demo-app-policy
+```
+The output string will look like below:
+
+```shell
+curl -X POST -H "X-Vault-Token: $(vault print token)" -H "X-Vault-Request: true" -d '{"policies":["vault-demo-app-policy"],"ttl":"0s","explicit_max_ttl":"0s","period":"0s","display_name":"","num_uses":0,"renewable":true,"type":"service","entity_alias":""}' http://127.0.0.1:8200/v1/auth/token/create
+```
```

Comment:
Is this part still needed if there is no warning from the previous command?

Reply:
Removed the note. I see the lastest Vault version in Dockerhub is 1.12.0. The post mentions Vaultt 1.11.3 in the tools list and the version is in a screenshot. I will qa again to get a fresh screenshot. Let me know if not required.

Reply:
A fresh screenshot would be awesome!

Reply:
Updated Vault screenshot and tools list.